parcels/particleset.py

```diff
@@ -96,11 +97,14 @@ def cptr(i):
 
             for i in range(size):
                 self.particles[i] = pclass(lon[i], lat[i], fieldset=fieldset, depth=depth[i], cptr=cptr(i), time=time[i])
+                # Set other Variables if provided
+                for kwvar in kwargs:
+                    setattr(self.particles[i], kwvar, kwargs[kwvar][i])
```

Comment:
Since kwargs gives a lot of freedom,  it would be nice to test here if kwvar is a particle attribute, with a if(hasattr)
or a try/except, with a nice print in case of error
(a difficulty for new users remains the reading of the error message when there is a problem)